### DISCUSSION TITLE
Google Adsense is not properly exposed when client side rendering.

### DISCUSSION THREAD

**Discussion by myriky** [vercel/next.js#11703] [Help]
# Bug report

## Describe the bug

Google Adsense is not properly exposed when client side rendering.
only first page by server side rendering showing well.

any ideas?

## To Reproduce

```html
<script async src="https://pagead2.googlesyndication.com/pagead/js/adsbygoogle.js" />
```
in next/head

```javascript
componentDidMount() {
    if (window) {
      (window.adsbygoogle = window.adsbygoogle || []).push({});
    }
}
```

```javascript
render() {
    return (
      <ins
        id="ga-ins"
        className="adsbygoogle"
        style={this.props.style}
        data-ad-client="ca-pub-blablabla"
        data-ad-format={this.props.format}
        data-ad-layout-key={this.props.layoutKey}
        data-ad-slot={this.props.slot}
        data-full-width-responsive={true}
      />
    );
}
```
## Expected behavior

![스크린샷 2020-04-07 오전 1 58 04](https://user-images.githubusercontent.com/581861/78584513-47d5fd80-7873-11ea-8034-a58cfc057d25.png)

## Screenshots

![스크린샷 2020-04-07 오전 12 54 49](https://user-images.githubusercontent.com/581861/78578400-73a0b580-786a-11ea-9aad-42fca1afd40e.png)

## System information

- OS: macOS
- Browser (if applies) Chrome
- Version of Next.js: 9.3.0

## Additional context

I've made a demo page. check out this pages.
[https://mask.shopping/adsense](https://mask.shopping/adsense)

**Comment by c0b41** [comment]
@myriky do you solved this problem?, i have exact problem too.

**Comment by myriky** [comment]
@c0b41 I closed the issue because it seemed to be work properly intermittent.

I couldn’t determine whether this issue is exactly related in Next.js or Google Adsense. 

I'll wait for the issue to be reopened

**Comment by ankit5999** [comment]
hey, could you please tell me how can I show ads in the next js applications? It shows blank at the place of ad-block.

**Reply by ankit5999** [reply]
I found a solution for this. 
Create a file using any name and call that where you want to show you ad.

Code:

import React from 'react';
import PropTypes from 'prop-types';

const initAd = () => {
  (window.adsbygoogle = window.adsbygoogle || []).push({});
};

class Adsencead extends React.Component {
  componentDidMount() {
    initAd();
  }

  shouldComponentUpdate(nextProps) {
    const { props: { path } } = this;
    return nextProps.path !== path;
  }

  componentDidUpdate() {
    initAd();
  }

  render() {
    const { children, className, path } = this.props;
    return (
      <div key={path} className={`adsense ${className}`}>
        <ins className="adsbygoogle"
     style={{display:"block"}}
     data-ad-client="" // put your adsence data
     data-ad-slot="" // put your adsence data
     data-ad-format="auto"
     data-full-width-responsive="true"></ins>
      </div>
    );
  }
}

Adsencead.propTypes = {
  path: PropTypes.string,
  className: PropTypes.string,
};

Adsencead.defaultProps = {
  className: '',
};

export default Adsencead;

**Comment by azokatheo** [comment]
i need help please

**Reply by ankit5999** [reply]
try this code once and let me know

**Comment by simra-aulakh** [comment]
hey , did u solve the issue of client side rendering ? 
let me know if u found the solution. 
Thanks

**Comment by yanghoxom** [comment]
just move `window.adsbygoogle = window.adsbygoogle || []).push({});` logic into a navigate event handler
like this
```
  const pathname = usePathname();
  const searchParams = useSearchParams();
  useEffect(() => {
    addAds();

    return () => {
      removeAds();
    };
  }, [pathname, searchParams]);
```
